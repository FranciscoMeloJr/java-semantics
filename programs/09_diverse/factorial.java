class main {
  int factorial(int y) {
    if (y == 1) {
      return 1;
    } else {
      return y*factorial(y-1);
    }
  }

  int read() {
    return Integer.parseInt(System.console().readLine());
  }

  void main(String[] args) {
    System.out.print("Input a natural numer: ");
    int n = read();
    System.out.println("Factorial of "+n+" is: "+factorial(n));
    System.out.print("Done!"+"\n");
  }
}
