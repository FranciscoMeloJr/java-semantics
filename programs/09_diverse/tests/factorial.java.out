Input a natural numer: Factorial of 5 is: 120
Done!
